arbor/simulation.cpp

```diff
@@ -466,7 +450,9 @@ time_type simulation_state::run(time_type tfinal, time_type dt) {
         foreach_cell(
             [&](cell_size_type i) {
                 PE(communication:enqueue:sort);
-                util::sort(pending_events_[i]);
+                // We just care about evt time not the ordering below that
```

Comment:
For reproducability when using different numbers of MPI ranks, we need to sort by the full event, not just event time.

Why does this matter? LIF cells are sensitive to the order that events due at the same time are delivered if the events have different weights w1 and w2, where w1 would cross the LIF threshold, and w2 would not.

Reply:
This was the root cause of a nasty reproducibility issue that took quite a lot of effort to track down.
In hindsight, we should have left a big warning comment in the code to educate future generations.

Reply:
Indeed that's a nasty corner case, especially as LIF is not used as often. I'll add the sorting on the full event (although it's costing us a bit).

Reply:
It's pretty easy to write a non-contrived `NET_RECEIVE` method that would also cause non-deterministic behavior in cable cells, unfortunately.

Reply:
Hi @bcumming,

while I think the problem wouldn't show up anymore with LIF cells as this is
currently in place 
```cxx
            const auto& event_lane = event_lanes[lid];
            // process all events at time t
            auto weight = 0.0;
            for (; event_idx < n_events && event_lane[event_idx].time <= time; ++event_idx) {
                weight += event_lane[event_idx].weight;
            }
```
but we cannot play the same trick on cable cells as the dynamics are way more open.

I added the big scary comment and rolled back the sorting.